sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/descriptor/MetricDescriptor.java

```diff
@@ -25,6 +27,14 @@ public static MetricDescriptor create(String name, String description, String un
     return new AutoValue_MetricDescriptor(name, description, unit);
   }
 
+  /** Constructs a metric descriptor for a given View + instrument. */
+  public static MetricDescriptor create(View view, InstrumentDescriptor instrument) {
+    final String name = (view.getName() == null) ? instrument.getName() : view.getName();
+    final String description =
```

Comment:
Nit: we don't generally make our local variables `final` unless there's a real reason for it.